### DISCUSSION TITLE
Not able to listen to WebSocket event after QR Modal (seems WS connection closed after modal pops up))

### DISCUSSION THREAD

**Discussion by jingxuan98** [WalletConnect/walletconnect-monorepo#2823] [Developer Support]
Output:

When click the connect walletConnect button, it triggers the signIn() by console.log("siginnn") then able to at least listen to the "display_uri" event but then the websocket seems its closing after this when i try to copy the URL and connect with `https://react-wallet.walletconnect.com/walletconnect`

![Screenshot 2023-06-28 at 12 57 53 PM](https://github.com/WalletConnect/walletconnect-monorepo/assets/43902263/bc80bb87-a83c-4299-8d1c-eca08815a3b6)

When trying to conenct the wallet via the wallet connect URL, i get these logs in my react web app terminal

![Screenshot 2023-06-28 at 1 06 05 PM](https://github.com/WalletConnect/walletconnect-monorepo/assets/43902263/d06bb7ed-619e-4e1b-aa36-db555fa894b9)

![Screenshot 2023-06-28 at 1 04 41 PM](https://github.com/WalletConnect/walletconnect-monorepo/assets/43902263/3c294224-b1f5-4cd3-a85a-72c54b94ccfe)

Any idea or solutions for this? 

GetWallet Provider part
```

    public async getWCProvider(): Promise<WalletConnectProvider> {
        this._walletConnectProvider = await EthereumProvider.init({
            projectId: 'project_ID',
            chains: [1, 5, 338, 80001],
            showQrModal: true,
            methods: [
                'eth_sendTransaction',
                'personal_sign',
                'eth_requestAccounts',
                'eth_signTypedData_v4',
              ],
            events: ['chainChanged', 'accountsChanged', 'connect', 'disconnect'],
        });
        return this._walletConnectProvider;
    }```

SignIn part
```
public async signIn(): Promise<WALLET_STATUS> {
        if (!this._walletConnectProvider) {
            this._walletConnectProvider = await this.getWCProvider()
        }
        await this._registerListeners()

        console.log("signinnnn")
        await this._walletConnectProvider.enable();

        return WALLET_STATUS.OK;
    }

```

Register WC part
```
private async _registerListeners() {
        if (!this._walletConnectProvider) {
            this._walletConnectProvider = await this.getWCProvider()
        }
        const rpc = await this._getRpc()
        const wc = this._walletConnectProvider

          try {
  
          wc.on("message", (args: any) => {
              console.log('message args', args)
  
          });
              
          wc.on("session_delete", (args: any) => {
              console.log('session_delete args', args)
  
          });
  
          wc.on("session_event", (args: any) => {
              console.log('session_event args', args)
  
          });
  
          wc.on("display_uri", (args: any) => {
              console.log('display_uri args', args)
  
              this._state.isConnected = true
          });
  
          
          wc.on('connect', (args: any) => {
              console.log('connect args', args)
  
              wc.modal?.closeModal();
  
          
          });
          wc.on('disconnect', (args: any) => {
              console.log('disconnect args', args)
  
    
          });
  
          wc.on('session_update', (args: any) => {
              console.log('session_update args', args)
  
      
          });
  
          wc.on('accountsChanged', async (accounts: string[]) => {
              this._state.accounts = accounts;
              if (accounts.length === 0) {
                  await this.signOut()
                  this.hookRouter.applyHooks([WALLET_HOOK.ACCOUNT_ON_DISCONNECT]);
              } else {
                  this.hookRouter.applyHookWithArgs(WALLET_HOOK.ACCOUNT_ON_CHANGE, accounts);
              }
              this._updateWalletStorageValue();
          });
  
          this._walletConnectProvider.on('chainChanged', async (args: string) => {
              console.log('chainChanged args', args)
  
              const chainId = 80001
              const id = ethers.utils.hexValue(chainId);
            
              this.hookRouter.applyHookWithArgs(WALLET_HOOK.CHAIN_ON_CHANGE, id);
          });
      } catch (e: any) {
          console.error(e)
      }
    }
```

**Comment by crypblizz8** [comment]
Hey @jingxuan98, we recommend putting chain 1 in `chains` and the rest through optionalChains

Also by any chance are you in Asia/China as you may need to use a VPN. Could you try that?

**Reply by jingxuan98** [reply]
Hmm tried both and not working... seems like my socket connection is closed ever since the QR modal pops up
